### PR TITLE
FlagMaps may be nil (prior to being initialized)

### PR DESCRIPTION
**Pull request by smarterclayton**
Do not panic if they haven't been set yet.

Fixes: #64090

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ func NewMapStringBool(m *map[string]bool) *MapStringBool {
 
 // String implements github.com/spf13/pflag.Value
 func (m *MapStringBool) String() string {
+	if m.Map == nil {
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool.go]: where are they set up and nil? Set() wouldn't work on one like that

**Comment by hzxuzhonghu** [comment]
This looks good, other than that should we also check if `m == nil`

**Comment by smarterclayton** [comment]
The check isn’t for m == nil, that would be coding error.  The bug is for maps that haven’t been initialized and don’t have a default.

**Comment by smarterclayton** [comment]
@liggitt see my last comment

**Comment by liggitt** [comment]
> @liggitt see my last comment

isn't m.Map being nil a coding error also? Set() fails. I'd expect String() to fail as well... since it can't return an error, isn't as panic better than silently appearing to succeed?

File: staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool.go [hunk 2]
```diff
@@ -34,6 +34,9 @@ type MapStringBool struct {
 // NewMapStringBool takes a pointer to a map[string]string and returns the
 // MapStringBool flag parsing shim for that map
 func NewMapStringBool(m *map[string]bool) *MapStringBool {
+	if *m == nil {
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool.go]: leaving this nil until Set is called for the first time was intentional (difference between empty (`--foo=`) and unset (no `--foo` flag at all)

if this is just to fix e2e help, can we initialize the flag correctly there?

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool.go]: This can not fix e2e help panic

**Comment by smarterclayton** [comment]
I see what you mean by usage.  This is ugly code and shouldn't be used like this (too hard to reason about).

Fixing `String()` to return empty string obeys the correct contract of a flag Value.

**Pull request merged**